spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/web/jetty/JettyMetricsAutoConfiguration.java

```diff
@@ -0,0 +1,87 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.metrics.web.jetty;
+
+
+import java.util.Collections;
+
+import io.micrometer.core.instrument.binder.jetty.JettyStatisticsMetrics;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.StatisticsHandler;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type;
+import org.springframework.boot.web.embedded.jetty.ConfigurableJettyWebServerFactory;
+import org.springframework.boot.web.embedded.jetty.JettyReactiveWebServerFactory;
+import org.springframework.boot.web.embedded.jetty.JettyServletWebServerFactory;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for {@link JettyStatisticsMetrics}.
+ *
+ * @author Even Holthe
+ * @since 2.1.0
+ */
+@ConditionalOnWebApplication
+@ConditionalOnClass({ JettyStatisticsMetrics.class, Server.class })
+@ConditionalOnProperty(prefix = "server.jetty", name = "publish-statistics", matchIfMissing = true)
+public class JettyMetricsAutoConfiguration {
+	private volatile StatisticsHandler statsHandler;
+
+	@Bean
+	@ConditionalOnMissingBean(JettyStatisticsMetrics.class)
+	public JettyStatisticsMetrics jettyMetrics() {
+		return new JettyStatisticsMetrics(this.statsHandler, Collections.emptyList());
+	}
+
+	@Bean
+	@ConditionalOnWebApplication(type = Type.SERVLET)
+	public WebServerFactoryCustomizer<JettyServletWebServerFactory> serverCapturingServletJettyCustomizer() {
+		return this::statisticsCustomizer;
+	}
+
+	@Bean
+	@ConditionalOnWebApplication(type = Type.REACTIVE)
+	public WebServerFactoryCustomizer<JettyReactiveWebServerFactory> serverCapturingReactiveJettyCustomizer() {
+		return this::statisticsCustomizer;
+	}
+
+	/**
+	 * Add a statistics handler to all Jetty's {@link Connector}s.
+	 *
+	 * @param server An already configured Jetty {@link Server}.
+	 */
+	private void configureStatistics(Server server) {
```

Comment:
I don't think we need this method comment. The method could be renamed to `addStatisticsHandlerToConnectors` if the intent isn't clear without it.